docs: adds Understanding Kubernetes Network Security  pillar page (#976)
cilium/cilium.io: src/posts/2026-06-07-understanding-kubernetes-network-security/index.md

## Encryption and Mutual Authentication

Encryption protects data in transit between workloads and nodes. Cilium supports transparent encryption using IPsec or WireGuard, allowing traffic between Cilium-managed endpoints, and in some configurations, host traffic to be encrypted without requiring application changes.
For stronger workload identity verification, Cilium also supports mutual authentication through its service mesh capabilities. This helps ensure that communication is not only encrypted but also tied to verified identities, which is especially important for zero-trust Kubernetes environments.

[Read More About Encryption](https://docs.cilium.io/en/latest/security/network/encryption/)

# 5\. The Anatomy of a Rule

A NetworkPolicy is a Kubernetes object, and its rule structure (the "anatomy") consists of four primary parts: Selector (who is this for), direction (which way is traffic moving), the peer, and the action.

## Subject Selectors and Identity

> Cilium Network Policies shift from volatile IP addresses to Identity-Based Security. Policies use Kubernetes metadata and labels known as Subject Selectors to define the enforcement point for both source and destination workloads (e.g., _role: frontend_ talking to _role: backend_). This logical abstraction translates into a simple numeric identity within the eBPF data path, enabling O(1) high-speed policy enforcement regardless of cluster scale.

Reviewer: ```suggestion
Cilium Network Policies shift from volatile IP addresses to Identity-Based Security. Policies use Kubernetes metadata and labels known as Subject Selectors to define the enforcement point for both source and destination workloads (e.g., _role: frontend_ talking to _role: backend_). This logical abstraction is pre-computed by the control plane into a single, cluster-wide Numeric Security Identity ($O(1)$ lookup complexity), allowing the eBPF kernel maps to execute high-speed packet filtering instantly.
```

## Default Deny Posture

A core element of a Zero Trust architecture, the Default Deny posture aggressively blocks all communication unless it is explicitly permitted by a policy. Teams can safely transition to this model by using Hubble (Cilium's observability layer) to visualize existing traffic and test policies in Audit Mode before moving to full enforcement.

## Ingress & Egress Semantics

Rules control the flow of traffic _into_ the workload (Ingress) and _out of_ the workload (Egress). For external communication, Cilium Egress Gateway can be used to secure egress access by routing traffic to specific external services through designated nodes, providing a stable, predictable source IP that satisfies traditional firewalls outside the cluster.

## Layer 7 Inspection

> This capability extends security beyond Layer 3/4 filtering to address the Layer 7 Blindness in Network Policies. It allows for fine-grained, API-aware security (e.g., DNS-aware rules) by enabling policies to inspect the application content. For instance, a policy can allow a _GET /public_ request while explicitly blocking a sensitive _POST /admin/shutdown_ command, strictly enforcing the Principle of Least Privilege.

Reviewer: ```suggestion
This capability extends security beyond Layer 3/4 filtering to address the Layer 7 blindness in network policies. It allows for fine-grained, API-aware security (e.g., DNS-aware rules) by enabling policies to inspect the application content. For instance, a policy can allow a _GET /public_ request while explicitly blocking a sensitive _POST /admin/shutdown_ command, strictly enforcing the Principle of Least Privilege.
```

[Read More About Cilium Network Policies](https://isovalent.com/blog/post/intro-to-cilium-network-policies/)

# 6\. Advanced Security Patterns

Cilium supports stronger patterns such as transparent encryption, mutual authentication, and host-layer segmentation to protect traffic and infrastructure without adding unnecessary operational complexity.

**Transparent Encryption (mTLS)**
Kubernetes lacks native pod-to-pod encryption, often requiring complex service mesh sidecars. Cilium solves this by providing **Transparent Encryption** for node-to-node traffic using technologies like IPsec or WireGuard, without requiring application changes. Recently, Cilium introduced **Native mTLS with ztunnel**, which unifies authentication and encryption directly within the Kubernetes data path, offering high-performance, identity-aware security for East-West traffic.

![mTLS Encryption](./images/m-tls-encryption.png)

> **\*Figure 5:** Transparent mTLS encryption and identity management for pod-to-pod communication powered by Cilium and eBPF.\*

Reviewer: Can you fix all the figures so they are just italics like in the other PR?

Author: Yes, have added Italics for all Figure Captions

**Host-Layer Segmentation**
Security strategies must also extend to the Node itself to prevent attackers who compromise the host from bypassing pod-level controls. **Cilium Host Policies** apply the same identity-aware security logic used for applications to secure the host network namespace, segmenting control plane traffic and SSH access.

> # 8\. Day 2 Operations: Observability, Auditing, and Compliance

Reviewer: ```suggestion
# 7\. Day 2 Operations: Observability, Auditing, and Compliance
```

## Auditing and Compliance

For regulated industries (FinTech, Healthcare), meeting compliance frameworks like SOC2, PCI-DSS, and GDPR requires to be detailed auditing. Cilium facilitates this by enforcing strict segmentation and generating detailed, immutable network flow logs. Tetragon further extends this capability by providing runtime security and visibility into system calls, supporting real-time threat protection and streamlined compliance reporting.

## Importance of Day 2 Operations

It's important to have Day 2 operations in mind when laying the groundwork for your software in Day 0 and Day 1, especially for cloud-native technologies. Having the right maintenance tools in place earlier on will help your organization avoid issues in the future.
Some common challenges during Day 2 operations include having trouble visualizing the performance of your software and difficulty integrating updates. Managing all the moving parts of your software is especially challenging with cloud-native systems such as Kubernetes, as they become increasingly more complex with scale.

> # 9\. Summary

Reviewer: ```suggestion
# 8\. Summary
```
